### ISSUE TITLE
transmission loss display failed

### ISSUE THREAD

**Issue by Vanillaholic** [patel999jay/Bellhop-ARLPY-ECED6575#4]
Sir, I'm here again. Recently, when I was drawing the transmission loss, the figure always failed to display. Have you ever encountered such a problem?🥹🥹🥹

<img width="976" alt="Image" src="https://github.com/user-attachments/assets/52417523-1ec0-496d-a390-9a237e7a78e3" />

**Comment by patel999jay** [comment]
i tried to replicate your issue, it should work fine with recommanded bokeh version, i have no issue : 

![Image](https://github.com/user-attachments/assets/31ec2956-b85d-4a0b-a982-e6cf7b938bbb)

MWE :

```python
ipython                                                                                                ─╯
Python 3.12.11 | packaged by Anaconda, Inc. | (main, Jun  5 2025, 08:06:15) [Clang 14.0.6 ]
Type 'copyright', 'credits' or 'license' for more information
IPython 9.1.0 -- An enhanced Interactive Python. Type '?' for help.
Tip: Run your doctests from within IPython for development and debugging. The special %doctest_mode command toggles a mode where the prompt, output and exceptions display matches as closely as possible that of the default Python interpreter.

In [1]: import arlpy.uwapm as pm
   ...: import arlpy.plot as plt
   ...: import numpy as np
   ...:
   ...:
   ...: env = pm.create_env2d()
   ...: pm.print_env(env)
   ...:
   ...: pm.plot_env(env, width=900)
   ...:
                name : arlpy
   bottom_absorption : 0.1
      bottom_density : 1600
    bottom_roughness : 0
   bottom_soundspeed : 1600
               depth : 25
        depth_interp : linear
           frequency : 25000
           max_angle : 80
           min_angle : -80
              nbeams : 0
            rx_depth : 10
            rx_range : 1000
          soundspeed : 1500
   soundspeed_interp : spline
             surface : None
      surface_interp : linear
            tx_depth : 5
   tx_directionality : None
                type : 2D

In [2]: !pip show bokeh
/usr/local/anaconda3/envs/arlpy-test/lib/python3.12/site-packages/IPython/core/interactiveshell.py:2645: UserWarning: You executed the system command !pip which may not work as expected. Try the IPython magic %pip instead.
  warnings.warn(
Name: bokeh
Version: 3.7.3
Summary: Interactive plots and applications in the browser from Python
Home-page: https://bokeh.org
Author: Bokeh Team
Author-email: info@bokeh.org
License: Copyright (c) Anaconda, Inc., and Bokeh Contributors
All rights reserved.
...
Location: /usr/local/anaconda3/envs/arlpy-test/lib/python3.12/site-packages
Requires: contourpy, Jinja2, narwhals, numpy, packaging, pandas, pillow, PyYAML, tornado, xyzservices
Required-by: arlpy

In [3]: rays = pm.compute_eigenrays(env)

In [4]: bathy = [
   ...:     [0, 30],    # 30 m water depth at the transmitter
   ...:     [300, 20],  # 20 m water depth 300 m away
   ...:     [1000, 25]  # 25 m water depth at 1 km
   ...: ]
   ...:

In [5]: ssp = [
   ...:     [ 0, 1540],  # 1540 m/s at the surface
   ...:     [10, 1530],  # 1530 m/s at 10 m depth
   ...:     [20, 1532],  # 1532 m/s at 20 m depth
   ...:     [25, 1533],  # 1533 m/s at 25 m depth
   ...:     [30, 1535]   # 1535 m/s at the seabed
   ...: ]

In [6]: env = pm.create_env2d(
   ...:     depth=bathy,
   ...:     soundspeed=ssp,
   ...:     bottom_soundspeed=1450,
   ...:     bottom_density=1200,
   ...:     bottom_absorption=1.0,
   ...:     tx_depth=15
   ...: )

In [7]: env['rx_range'] = np.linspace(0, 1000, 1001)
   ...: env['rx_depth'] = np.linspace(0, 30, 301)

In [8]: tloss = pm.compute_transmission_loss(env)
   ...: pm.plot_transmission_loss(tloss, env=env, clim=[-60,-30], width=900)
```

also to get ride of the `Bokeh` warnings for now; you can do this, find your `alrpy.plot.scatter()` and replace that function with below (only temporary fix, for now): 

```python
def scatter(x, y, marker='.', filled=False, size=6, color=None, title=None, xlabel=None, ylabel=None, xlim=None, ylim=None, xtype='auto', ytype='auto', width=None, height=None, legend=None, hold=False, interactive=None):
    """Plot a scatter plot.

    :param x: x data
    :param y: y data
    :param color: marker color (see `Bokeh colors`_)
    :param marker: point markers ('.', 'o', 's', '*', 'x', '+', 'd', '^')
    :param filled: filled markers or outlined ones
    :param size: marker size
    :param title: figure title
    :param xlabel: x-axis label
    :param ylabel: y-axis label
    :param xlim: x-axis limits (min, max)
    :param ylim: y-axis limits (min, max)
    :param xtype: x-axis type ('auto', 'linear', 'log', etc)
    :param ytype: y-axis type ('auto', 'linear', 'log', etc)
    :param width: figure width in pixels
    :param height: figure height in pixels
    :param legend: legend text
    :param interactive: enable interactive tools (pan, zoom, etc) for plot
    :param hold: if set to True, output is not plotted immediately, but combined with the next plot

    >>> import arlpy.plot
    >>> import numpy as np
    >>> arlpy.plot.scatter(np.random.normal(size=100), np.random.normal(size=100), color='blue', marker='o')
    """
    global _figure, _color
    if _figure is None:
        _figure = _new_figure(title, width, height, xlabel, ylabel, xlim, ylim, xtype, ytype, interactive)
    x = _np.array(x, ndmin=1, dtype=_np.float64, copy=False)
    y = _np.array(y, ndmin=1, dtype=_np.float64, copy=False)
    if color is None:
        color = _colors[_color % len(_colors)]
        _color += 1
    kwargs = {'size': size, 'line_color': color}
    if filled:
        kwargs['fill_color'] = color
    if legend is not None:
        kwargs['legend_label'] = legend
    if marker == '.':
        kwargs['size'] = kwargs['size']/2
        kwargs['fill_color'] = color
        _figure.circle(x, y, **kwargs)
    elif marker == 'o':
        _figure.scatter(x, y, marker='circle', **kwargs) # This
    elif marker == 's':
        _figure.square(x, y, **kwargs)
    elif marker == '*':
        _figure.scatter(x, y, marker='asterisk', **kwargs) # This
    elif marker == 'x':
        _figure.x(x, y, **kwargs)
    elif marker == '+':
        _figure.cross(x, y, **kwargs)
    elif marker == 'd':
        _figure.diamond(x, y, **kwargs)
    elif marker == '^':
        _figure.triangle(x, y, **kwargs)
    elif marker is not None:
        _warnings.warn('Bad marker type: '+marker)
    if not hold and not _hold:
        _show(_figure)
        _figure = None
```

**Comment by Vanillaholic** [comment]
@patel999jay Thank you very very much，there may be some parameters I
set were wrong.Arlpy is different with Matlab🥹，once the argument is wrong the figure is wrong![image](https://github.com/user-attachments/assets/011646c2-68b9-4fc0-bea4-1f7a586936de)

**Comment by Vanillaholic** [comment]
@patel999jay Thank you very very much，there may be some parameters I
set were wrong.
![image](https://github.com/user-attachments/assets/66b7848d-4d91-427b-99dd-04b0ef00955c)
Arlpy is different with Matlab🥹，once the argument is wrong the figure is wrong
![image](https://github.com/user-attachments/assets/79e17f7a-b3ad-4f59-84fb-f8a9a1454724)
